contract/lib.rs

```diff
@@ -0,0 +1,331 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod errors;
+
+#[cfg(test)]
+pub mod tests;
+
+#[ink::contract]
+mod syndeo {
+    use crate::errors::ContractError;
+
+    use ink::storage::Mapping;
+    use ink_prelude::vec::Vec;
+
+    #[derive(PartialEq, Debug, Eq, Clone, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub struct RewardsSummary {
+        assigned_points: u64,
+        members_awarded: u64,
+        funds: Balance,
+    }
+
+    #[ink(event)]
+    pub struct AdminChanged {
+        admin: AccountId,
+    }
+
+    #[ink(event)]
+    pub struct NewMember {
+        member: AccountId,
+    }
+
+    #[ink(event)]
+    pub struct MemberDeleted {
+        member: AccountId,
+    }
+
+    #[ink(event)]
+    pub struct Award {
+        sender: AccountId,
+        recipient: AccountId,
+        amount: u64,
+    }
+
+    #[ink(event)]
+    pub struct RewardGranted {
+        recipient: AccountId,
+        reward: Balance,
+        points: u64,
+    }
+
+    #[ink(storage)]
+    pub struct Syndeo {
+        pub admin: AccountId,
+        pub members: Vec<AccountId>,
+        pub points_by_sender: Mapping<AccountId, u64>,
+        pub senders: Vec<AccountId>,
+        pub points_by_recipient: Mapping<AccountId, u64>,
+        pub recipients: Vec<AccountId>,
+        pub total_points: u64,
+        pub max_points_per_sender: u64,
+    }
+
+    impl Syndeo {
+        #[ink(constructor)]
+        pub fn new(max_points_per_sender: Option<u64>) -> Self {
+            let caller = Self::env().caller();
+            let mut members = Vec::new();
+            members.push(caller);
+
+            Self {
+                admin: caller,
+                members,
+                points_by_sender: Mapping::default(),
+                senders: Vec::new(),
+                points_by_recipient: Mapping::default(),
+                recipients: Vec::new(),
+                total_points: 0,
+                max_points_per_sender: max_points_per_sender.unwrap_or(10),
+            }
+        }
+
+        #[ink(message)]
+        pub fn set_admin(&mut self, new_admin: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if !self.is_member(&new_admin) {
+                self.add_member(new_admin)?;
+            }
```

Comment:
Admin might not necessarily be a member, I think we can remove this part

Reply:
Are you saying that an admin can be external to the DAO? Sounds weird..